solr/core/src/java/org/apache/solr/search/facet/TermFrequencyOfFrequenciesAgg.java

```diff
@@ -0,0 +1,73 @@
+package org.apache.solr.search.facet;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.apache.lucene.queries.function.ValueSource;
+import org.apache.solr.search.FunctionQParser;
+import org.apache.solr.search.SyntaxError;
+import org.apache.solr.search.ValueSourceParser;
+
+public class TermFrequencyOfFrequenciesAgg extends SimpleAggValueSource {
+  private final int termLimit;
+
+  public TermFrequencyOfFrequenciesAgg(ValueSource vs, int termLimit) {
+    super("termfreqfreq", vs);
+
+    this.termLimit = termLimit;
+  }
+
+  @Override
+  public SlotAcc createSlotAcc(FacetContext fcontext, int numDocs, int numSlots) {
+    return new TermFrequencySlotAcc(getArg(), fcontext, numSlots, termLimit);
+  }
+
+  @Override
+  public FacetMerger createFacetMerger(Object prototype) {
+    return new Merger(termLimit);
+  }
+
+  public static class Parser extends ValueSourceParser {
+    @Override
+    public ValueSource parse(FunctionQParser fp) throws SyntaxError {
+      ValueSource vs = fp.parseValueSource();
+
+      int termLimit = Integer.MAX_VALUE;
+      if (fp.hasMoreArguments()) {
+        termLimit = fp.parseInt();
+      }
+
+      return new TermFrequencyOfFrequenciesAgg(vs, termLimit);
+    }
+  }
+
+  private static class Merger extends FacetMerger {
+    private final TermFrequencyCounter result;
+
+    public Merger(int termLimit) {
```

Comment:
is `termLimit` not used now?

Reply:
oh yes, it used to be on TermFrequencyCounter but it's moved now